lib/puppet/configurer.rb

```diff
@@ -80,6 +81,8 @@ def retrieve_catalog(query_options)
         @cached_catalog_status = 'explicitly_requested'
       elsif @pluginsync_failed
         @cached_catalog_status = 'on_pluginsync_failure'
+      elsif @server_connection_failed
+        @cached_catalog_status = 'on_server_connection_failure'
```

Comment:
I have a similar concern with a different status as mentioned in https://github.com/puppetlabs/puppet/pull/8328/files#r500659288 for `on_pluginsync_failure`